pallets/gear/src/internal.rs

```diff
@@ -602,6 +688,70 @@ where
         });
     }
 
+    /// Sends user message, once delay reached.
+    pub(crate) fn send_user_message_after_delay(message: StoredMessage) {
+        // Converting payload into string.
+        //
+        // Note: for users, trap replies always contain
+        // string explanation of the error.
+        let message = match message.exit_code() {
+            Some(0) | None => message,
+            _ => message
+                .with_string_payload::<ExecutionErrorReason>()
+                .unwrap_or_else(|e| {
+                    log::debug!("Failed to decode error to string");
+                    e
+                }),
+        };
+
+        // Taking data for funds manipulations.
+        let from = <T::AccountId as Origin>::from_origin(message.source().into_origin());
+        let to = <T::AccountId as Origin>::from_origin(message.destination().into_origin());
+        let value = message.value().unique_saturated_into();
+
+        // If gas limit can cover threshold, message will be added to mailbox,
+        // task created and funds reserved.
+        let expiration = if GasHandlerOf::<T>::exists(message.id()) {
+            // Querying gas limit. Fails in cases of `GasTree` invalidations.
+            let gas_limit = GasHandlerOf::<T>::get_limit(message.id())
```

Comment:
It will panic in the case I described upper with gas limit being less than mailbox threshold. That's because no detached node will be created, so the query will panic